### DISCUSSION TITLE
refetch loader

### DISCUSSION THREAD

**Discussion by VladyPoim** [TanStack/router#278] [Q&A]
Hi,

I'm trying to refresh loader data after navigation with new search params by using hook useNavigation. Unfortunately, it was not possible to update the loader data in this way. Perhaps there are other ways to refetch data?

**Comment by jintaokoong** [comment]
I'm wondering the same exact thing. It'd be great if by changing the search param, loader function will be triggered automatically. One way of solving this is for the time being, would be pulling a data fetching library like react query, passing the search params from useSearch into useQuery to trigger a refetch.

**Comment by MadadiSaeed** [comment]
This is a quote from react router website:

> By default, route loaders are called for **new or changed routes in the route hierarchy that resulted from a navigation**.

Specifically route's path. So i think it is better react router add a flag to useNavigation's navigate function to force router reload required loader. I am facing this problem in my react app and I had to add additional lines of code to work around this.

**Comment by mvacha** [comment]
I'm evaluating TanStack Router as a potential replacement for the React Router we currently use, and I have run into this issue too. React Router calls loader when search parameters change, but TanStack does not 😔.

This is my workaround, but I think it should be an option on the route or the useNavigate hook.
```javascript
//HACK: force loader to run when search params change
const search = useSearch();
const router = useRouter();
useEffect(() => {router.invalidate()}, [search, router]);
```

**Reply by tannerlinsley** [reply]
Can you show me an example, or your code please?

**Comment by tannerlinsley** [comment]
As of the latest version, it’s impossible to use search params in a loader without passing them through the loader context. Anything you pass through the loader context is watched for changes. When they change, your loader will rerun.

**Reply by tannerlinsley** [reply]
It’s very similar to Reacts dependency array tracking

**Reply by mvacha** [reply]
Thanks for a prompt response, I'm on 0.0.1-beta.192, so I'll look into loader context and migrate to 194.

**Reply by mvacha** [reply]
Got it working  :) 

BTW: Is it a feature or a bug that the search property passed to the loaderContext is not validated?

**Reply by tannerlinsley** [reply]
Hmmm… sounds like a bug

**Comment by bruceharrison1984** [comment]
Is there any feasible way of doing this, short of navigating back to the same route with some kind of random search param to essentially force rerunning of the loader?

My use case is essentially adding an item to a list, and I'd prefer not to have a full postback or duplicating the loader call on the page itself.

*edit*
It looks like navigating back to the same route has the effect I was looking for.

```ts
const navigate = Route.useNavigate();
...
const submitForm = () => {
  navigate({ to: '.' });
}
```

**Reply by BrendanC23** [reply]
You could also do `router.invalidate()`.
